lib/ansible/modules/network/junos/junos_vlans.py

```diff
@@ -0,0 +1,278 @@
+        type: int
+        required: true
+      name:
+        description:
+          - Name of VLAN.
+        type: str
+        required: true
+      description:
+        description:
+          - Text description of VLANs
+        type: str
+  state:
+    description:
+      - The state the configuration should be left in.
+    type: str
+    choices:
+      - merged
+      - replaced
+      - overridden
+      - deleted
+    default: merged
+"""
+
+EXAMPLES = """
+# Using merged
+#############
+
+# Before State
+# ------------
+#
+# admin# show vlans
+# vlan-2 {
+#     vlan-id 2;
+# }
+# vlan-3 {
+#     vlan-id 3;
+# }
+
+- name: Merge JUNOS vlan
+  junos_vlans:
+    config:
+      - name: vlan-1
+        vlan-id: 1
+  state: merged
+
+# After State
+# -----------
+#
+# admin# show vlans
+# vlan-1 {
+#     vlan-id 1;
+# }
+# vlan-2 {
+#     vlan-id 2;
+# }
+# vlan-3 {
+#     vlan-id 3;
+# }
+
+
+# Using replaced
+################
+
+# Before State
+# ------------
+#
+# admin# show vlans
+# vlan-1 {
+#     vlan-id 1;
+# }
+# vlan-2 {
+#     vlan-id 2;
+# }
+# vlan-3 {
+#     vlan-id 3;
+# }
+
+- name: Replace JUNOS vlan
+  junos_vlans:
+    config:
+      - name: vlan-1
+        vlan-id: 10
+      - name: vlan-3
+        vlan-id: 30
+  state: replaced
+
+# After State
+# -----------
+#
+# admin# show vlans
+# vlan-1 {
+#     vlan-id 10;
+# }
+# vlan-2 {
+#     vlan-id 2;
+# }
+# vlan-3 {
+#     vlan-id 30;
+# }
+
+
+# Using overridden
+##################
+
+# Before State
+# ------------
+#
+# admin# show vlans
+# vlan-1 {
+#     vlan-id 1;
+# }
+# vlan-2 {
+#     vlan-id 2;
+# }
+# vlan-3 {
+#     vlan-id 3;
+# }
+
+- name: Override JUNOS vlan
+  junos_vlans:
+    config:
+      - name: vlan-4
+        vlan-id: 100
+      - name: vlan-2
+        vlan-id: 200
+  state: overridden
+
+# After State
+# -----------
+#
+# admin# show vlans
+# vlan-2 {
+#     vlan-id 200;
+# }
+# vlan-4 {
+#     vlan-id 100;
+# }
+
+
+#Using deleted
+##############
+
+# Before State
+# ------------
+#
+# admin# show vlans
+# vlan-1 {
+#     vlan-id 1;
+# }
+# vlan-2 {
+#     vlan-id 2;
+# }
+# vlan-3 {
+#     vlan-id 3;
+# }
+
+- name: Delete JUNOS vlan
+  junos_vlans:
+    config:
+      - name: vlan-1
+  state: deleted
+
+# After State
+# -----------
+#
+# admin# show vlans
+# vlan-2 {
+#     vlan-id 2;
+# }
+# vlan-3 {
+#     vlan-id 3;
+# }
+"""
+
+RETURN = """
+before:
+  description: The configuration prior to the model invocation.
+  returned: always
+  type: str
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+after:
+  description: The resulting configuration model invocation.
+  returned: when changed
+  type: str
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: always
+  type: list
+  sample: ['command 1', 'command 2', 'command 3']
```

Comment:
since xml list is returned this can be modified to `sample: ['xml 1', 'xml 2', 'xml 3']`